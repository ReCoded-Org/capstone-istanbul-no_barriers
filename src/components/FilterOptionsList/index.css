:root {
    --main-filter-option-bg-color: #E7E7E7;
    --main-filter-option-text-color: #00374d;
    --orange-link: #f2c94c;
  }

  .filterOptionsList {
      background-color: var(--main-filter-option-bg-color);
      color: var(--main-filter-option-text-color);
  }

  .filterOption:not(:last-of-type) {
      border-bottom: 1px solid var(--main-filter-option-text-color);
  }

  .filterOptionPlusSign {
      margin-left: 1rem;
  }