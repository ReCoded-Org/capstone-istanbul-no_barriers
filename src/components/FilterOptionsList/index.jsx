import React from "react";

// This renders the filter options of a given filter as parameter.
// It is called in AdvancedFilters component
const FilterOptionsList = ({ filter }) => {
  return (
    <div className="filterOptionsList">
      {filter.filterParameters.map((option) => {
        return (
          <div className="filterOption d-flex">
            <span className="filterOptionPlusSign">+</span>
            <p className="filterOptionTitle">{option}</p>
          </div>
        );
      })}
    </div>
  );
};

export default FilterOptionsList;
