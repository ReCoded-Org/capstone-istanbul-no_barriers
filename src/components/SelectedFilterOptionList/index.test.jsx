import React from "react";
import renderer from "react-test-renderer";
import FilterOptionsList from "./index";
import { Provider } from "react-redux";
import store from "../../redux/store/store";

const selectedOptions = ["Ankara", "Istanbul"];

it("SelectedFilterOptionList renders correctly", () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <FilterOptionsList selectedOptions={selectedOptions} />
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
