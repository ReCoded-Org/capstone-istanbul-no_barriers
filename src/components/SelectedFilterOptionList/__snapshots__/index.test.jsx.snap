// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectedFilterOptionList renders correctly 1`] = `
<div
  className="filteredOptionsList"
>
  <div
    className="selectedOption d-flex justify-content-between backgroundLight"
  >
    <p
      className="selectedOptionTitle"
    >
      Ankara
    </p>
    <span
      className="selectedOptionDelete"
    >
      X
    </span>
  </div>
  <div
    className="selectedOption d-flex justify-content-between backgroundDark"
  >
    <p
      className="selectedOptionTitle"
    >
      Istanbul
    </p>
    <span
      className="selectedOptionDelete"
    >
      X
    </span>
  </div>
</div>
`;
